python/lsst/cp/pipe/utilsEfd.py

```diff
@@ -257,41 +315,163 @@ def parseMonochromatorStatus(self, data, dateStr):
         wavelength : `float`
             Monochromator commanded peak.
         """
-        dateValue = Time(dateStr, format='isot', scale='tai')
-        # Table is now sorted on "time", which is in UTC.
+        result, _ = self.searchResults(data, dateStr)
+        myTime = result["private_sndStamp"]
+        return myTime.strftime("%Y-%m-%dT%H:%M:%S.%f"), result['wavelength']
 
-        # Check that the date we want to consider is contained in the
-        # EFD data.
-        if data["time"][0] > dateValue or data["time"][-1] < dateValue:
-            raise RuntimeError("Requested date is outside of data range.")
+    def getEfdElectrometerData(self, dataSeries=None, dateMin=None, dateMax=None):
+        """Retrieve Electrometer data from the EFD.
 
-        # Binary search through the EFD entries in date, until the
-        # most recent monochromator state update prior to the spectrum
-        # in question is found.
-        low = 0
-        high = len(data)
-        idx = (high + low) // 2
-        found = False
-        iteration = 0
-        while not found:
-            if idx < 0 or idx > len(data) or iteration > 10:
-                raise RuntimeError("Search for date failed?")
+        Parameters
+        ----------
+        dataSeries : `str`, optional
+            Data series to request from the EFD.
+        dateMin : `str`, optional
+            Minimum date to retrieve from EFD.
+        dateMax : `str`, optional
+            Maximum date to retrieve from EFD.
 
-            myTime = data["private_sndStamp"][idx]
+        Returns
+        -------
+        results : `astropy.table.Table`
+            The table of results returned from the EFD.
+        """
+        # All electrometer data gets written to the same series.
+        defaultSeries = "lsst.sal.Electrometer.logevent_intensity"
```

Comment:
Okay this is where it gets very dicey.  What if we have comcam and latiss calibrations both going???

Reply:
They're distinguished by the `salIndex`.  I've updated the docstring in `parseElectrometerStatus` to list the values for the two telescopes.